routes/tweet.js

```diff
@@ -1,32 +1,270 @@
 const router = require("express").Router();
 let Tweet = require("../models/tweet.model");
+let User = require("../models/user.model");
+var ObjectID = require('mongodb').ObjectID;
 
-router.route("/").get((req, res) => {
-  Tweet.find()
+router.route("/getUserTweets").get((req, res) => {
+
+
+
+  Tweet.find(req.query).sort('-createdAt')
+    .then((tweets) => res.json(tweets))
+    .catch((err) => res.status(400).json("Error: " + err));
+});
+
+router.route("/getTweet").get((req, res) => {
+
+  Tweet.find({ _id: req.query.tweetUUID })
     .then((tweets) => res.json(tweets))
     .catch((err) => res.status(400).json("Error: " + err));
 });
 
+
+
+
+/**
+ * Called when making a retweet
+ * 
+ * First creates the retweet as a normal tweet. Then, if successfully made, updates
+ * the user who made the retweet with the fact that they made a retweet
+ * 
+ * Then, if this succeeds, updates the retweeted tweet with the id of the new retweet
+ * 
+ * 
+ * Nested because it needs to be atomic.
+ * 
+ * 
+ * Retweet with comment only
+ */
+router.route("/addRetweet").post((req, res) => {
+
+
+  const rt = { userID: req.body.userID, tweetBody: req.body.tweetBody, numOfLikes: 0, numOfRetweetsWithComment: 0, numOfRetweetsWithNoComment: 0 };
+
+
+  const retweetUserID = (new ObjectID(req.body.userID));
+  const tweetUUID = (new ObjectID(req.body.tweetUUID));
+
+  const newRetweet = new Tweet(rt);
+
+
+  newRetweet
+    .save()
+    .then(() => {
+
+
+      User.update(
+        { _id: retweetUserID }, //arguments object
+        { $push: { retweets: { tweetUUID: newRetweet._id } } }
+      ).then(() => {
+
+        Tweet.update(
+          { _id: tweetUUID },
+          { $push: { retweetedWithComment: { tweetUUID: newRetweet._id, userUUID: newRetweet.userID } }, $inc: { numOfRetweetsWithComment: 1 } }
```

Comment:
So this route only is for retweets with comments so maybe the route name should reflect this instead of being /addRetweet. Maybe like /addCommentedRetweet or something